debian/rules

```diff
@@ -44,6 +45,9 @@ ifeq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
 	apparmor_parser -QTK ./debian/usr.bin.messaging-app
 endif
 
+override_dh_auto_install:
+	dh_auto_install -Pdebian/tmp
+
```

Comment:
Is it necessary to do this? It's causing the build to fail in CI with:
```
dh_auto_install: -P was specified, but multiple packages would be acted on (messaging-app,messaging-app-autopilot).
debian/rules:49: recipe for target 'override_dh_auto_install' failed
make[1]: *** [override_dh_auto_install] Error 255
```

Reply:
Ah, in fact I tested only the configuration where the `messaging-app-autopilot` package was *not* built. So, the reason why I added it is that when the building of the autopilot package is disabled, the project consists of a single package. In that case, the files do not get installed into `debian/tmp`, but under `debian/messaging-app`, and then the build fails because dh_install cannot find any of the files listed in the `messaging-app.install` file.

I can fix it with adding an `if` somewhere, but I wonder if there's a better way to avoid all these complications.

Reply:
I don't see any valid reason to not build the `-autopilot` package though. Building it does not require running the tests, and therefore tying this to `nocheck` doesn't make sense. You could _avoid all these complexities_ by simply not having introduced them in the first place. :)

Reply:
I didn't introduce any complexities, see also Lionel's comment above (or try it yourself): currently, the project does not build with crossbuilder, because the `messaging-app-autopilot` files are not generated when crossbuilding.

Reply:
Well, why are they not being installed? What exactly is not being generated? It's just a python package and doesn't require anything special to install. The changes introduced here are surely more complex than simply fixing whatever it is that is preventing them to be installed.

Reply:
There are not being installed because the `tests/` directory is not being entered if cmake is invoked with `-DENABLE_TESTS=OFF`. I've now reworked the patch to make it so that that flag only controls the *execution* of the tests, and make autopilot tests always installed.

But that's just to make you happy :-) I believe that we do have a case for not wanting the test installation to happen (the fact that the `noinsttest` option exists shows that other people consider this worth, too).

Another (though not strictly related) consideration is that we should probably remove all the autopilots tests altogether, since AFAIK they haven't been run in ages and most likely don't work anymore.